snapcraft/internal/parts.py

```diff
@@ -117,6 +124,25 @@ def get_part(self, part_name, full=False):
                 remote_part.pop(key)
         return remote_part
 
+    def matches_for(self, part_match, max_len=0):
```

Comment:
This method is only ever used within this class though, no? Do you envision it being used by others?